src/umockdev.vala

```diff
@@ -644,6 +644,10 @@ public class Testbed: GLib.Object {
             return;
         }
 
+        // Generate the removal event (must be called before the directory
+        // structure is removed).
+        uevent(syspath, "remove");
```

Comment:
Please move this below the next paragraph which gets/checks the subsystem. When calling this on invalid devices, there shouldn't be a `remove` event.